edge/test/integration/device/device_test.go

```diff
@@ -41,134 +70,156 @@ var DeviceTW dttype.Device
+			Expect(TokenClient.Error()).NotTo(HaveOccurred())
 			Client.Disconnect(1)
 		})
 
 		It("TC_TEST_EBUS_3: Sending data to device twin module", func() {
-			if Token_client = Client.Publish(DeviceTwinUpdate, 0, false, "messagetoDevice_Twin_update"); Token_client.Wait() && Token_client.Error() != nil {
-				common.Failf("client.Publish() Error is %s", Token_client.Error())
+			if TokenClient = Client.Publish(DeviceTwinUpdate, 0, false, "messagetoDevice_Twin_update"); TokenClient.Wait() && TokenClient.Error() != nil {
+				common.Failf("client.Publish() Error is %s", TokenClient.Error())
 			} else {
 				common.InfoV6("client.Publish Success !!")
 			}
-			Expect(Token_client.Error()).NotTo(HaveOccurred())
+			Expect(TokenClient.Error()).NotTo(HaveOccurred())
 			Client.Disconnect(1)
 		})
 
 		It("TC_TEST_EBUS_4: Sending data to membership module", func() {
-			if Token_client = Client.Publish(DeviceMembershipUpdate, 0, false, "messagetoDevice_Membership_update"); Token_client.Wait() && Token_client.Error() != nil {
-				common.Failf("client.Publish() Error is %s", Token_client.Error())
+			if TokenClient = Client.Publish(DeviceMembershipUpdate, 0, false, "messagetoDevice_Membership_update"); TokenClient.Wait() && TokenClient.Error() != nil {
+				common.Failf("client.Publish() Error is %s", TokenClient.Error())
 			} else {
 				common.InfoV6("client.Publish Success !!")
 			}
-			Expect(Token_client.Error()).NotTo(HaveOccurred())
+			Expect(TokenClient.Error()).NotTo(HaveOccurred())
 			Client.Disconnect(1)
 		})
 
 		It("TC_TEST_EBUS_5: Sending data to device module", func() {
-			if Token_client = Client.Publish(DeviceUpload, 0, false, "messagetoDevice_upload"); Token_client.Wait() && Token_client.Error() != nil {
-				common.Failf("client.Publish() Error is %s", Token_client.Error())
+			if TokenClient = Client.Publish(DeviceUpload, 0, false, "messagetoDevice_upload"); TokenClient.Wait() && TokenClient.Error() != nil {
+				common.Failf("client.Publish() Error is %s", TokenClient.Error())
 			} else {
 				common.InfoV6("client.Publish Success !!")
 			}
-			Expect(Token_client.Error()).NotTo(HaveOccurred())
+			Expect(TokenClient.Error()).NotTo(HaveOccurred())
 			Client.Disconnect(1)
 		})
 
 		It("TC_TEST_EBUS_6: change the device status to online from eventbus", func() {
 			var message DeviceUpdate
 			message.State = "online"
+			topic := dtcommon.DeviceETPrefix + DeviceIDN + dtcommon.DeviceETStateUpdateSuffix + "/result"
 			body, err := json.Marshal(message)
 			if err != nil {
 				common.Failf("Marshal failed %v", err)
 			}
-
-			if Token_client = Client.Publish(dtcommon.DeviceETPrefix+DeviceIDN+dtcommon.DeviceETStateUpdateSuffix, 0, false, body); Token_client.Wait() && Token_client.Error() != nil {
-				common.Failf("client.Publish() Error is %s", Token_client.Error())
-			} else {
-				common.InfoV6("client.Publish Success !!")
+			Eventually(func() string {
+				Devicedb = GetDeviceFromDB(DeviceIDN)
```

Comment:
Is it necessary that e directly get the device from DB ? Can we use deviceTwin MQTT API's to get the device ?

Reply:
can be done...changed it..